src/Symfony/Component/DependencyInjection/Argument/ServiceLocator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class ServiceLocator
```

Comment:
Maybe we could make it implement the ArrayAccess interface. It would allow consume interfaces to use `array|ArrayAccess` as a type hint and use it like an array.